### PR TITLE
Async/Await support and types refactor of Nightwatch CDP commands.

### PR DESCRIPTION
**Pull request by garg3133**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## nightwatch ([<kbd>unpkg</kbd>](https://unpkg.com/browse/nightwatch@latest/))
was missing the following properties:
1. createClient
1. client
1. cli
1. runTests
1. CliRunner

<details>
<summary>as well as these 2 other properties...</summary>
<p>initClient, runner</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 352518a6a448ae5d2ac893190898fcc9bc2af9be
</p>

**Comment by garg3133** [comment]
There are two issues due to which the tests are failing:
* It is not allowing the use of non-arrow functions and the callback to `describe` should be a non-arrow function.
* It is not considering the type `Awaitable<T, V> = Omit<T, "then"> & PromiseLike<V>;` to be a promise.

**Review by vaibhavsingh97 — APPROVED** [review]
LGTY

**Review by vaibhavsingh97 — APPROVED** [review]

File: types/nightwatch/test/nightwatch-tests.ts [hunk 1]
```diff
@@ -284,6 +284,9 @@ const testPage = {
             .assert.title('Google') // deprecated
             .assert.titleEquals('Google') // new in 2.0
             .assert.visible('@searchBar')
+            .assert.not.titleContains('DuckDuckGo')
+            // @ts-expect-error
+            .assert.not.not.elementPresent('@searchbar')
```
Comment (weswigham) [inline types/nightwatch/test/nightwatch-tests.ts:289]: This error is going to make all the following chained items into `any`, and thus not actually test anything. You should make this its own chain.

**Review by weswigham — CHANGES_REQUESTED** [review]

**Pull request merged**